lib/ts/framework/express/framework.ts

```diff
@@ -0,0 +1,171 @@
+/* Copyright (c) 2021, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ * This software is licensed under the Apache License, Version 2.0 (the
+ * "License") as published by the Apache Software Foundation.
+ *
+ * You may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+import type { Request, Response, NextFunction } from "express";
+import type { HTTPMethod } from "../../types";
+import { normaliseHttpMethod } from "../../utils";
+import { BaseRequest } from "../request";
+import { BaseResponse } from "../response";
+import {
+    setCookieForServerResponse,
+    setHeaderForExpressLikeResponse,
+    getCookieValueFromIncomingMessage,
+    getHeaderValueFromIncomingMessage,
+    assertThatBodyParserHasBeenUsedForExpressLikeRequest,
+} from "../utils";
+import type { Framework } from "../types";
+import SuperTokens from "../../supertokens";
+import type { SessionContainerInterface } from "../../recipe/session/types";
+
+export class ExpressRequest extends BaseRequest {
+    private request: Request;
+    private parserChecked: boolean;
+
+    constructor(request: Request) {
+        super();
+        this.original = request;
+        this.request = request;
+        this.parserChecked = false;
+    }
+
+    getKeyValueFromQuery = async (key: string): Promise<string | undefined> => {
+        if (!this.parserChecked) {
```

Comment:
You don't need to add this if block here since this function does not use req.body